packages/snap/src/core/utils/errors.ts

```diff
@@ -88,3 +49,23 @@ export function isSnapRpcError(error: Error): boolean {
   ];
   return errors.some((errType) => error instanceof errType);
 }
+
+export const withCatchAndThrowSnapError = async <ResponseT>(
+  scope: string,
```

Comment:
nit: the `withCatchAndThrowSnapError` accepts a `scope` parameter that is only used for logging... I'd remove that if that's the only reason you have it there. The caller knows what function failed so if that's needed, the Snap environment (or extension) should be the one logging it IMO